### PR TITLE
dydx price feed used to get dai to usd rate

### PR DESCRIPTION
**Pull request by jchittoda**
still some previous tests are failing

### CODE REVIEW

File: src/BCdpManager.t.sol [hunk 1]
```diff
@@ -158,6 +158,17 @@ contract FakeOSM {
     }
 }
 
+contract FakeDaiToUsdPriceFeed {
+    uint price = 1e18;
+    function setPrice(uint newPrice) public {
+        price = newPrice;
+    }
+
+    function getMarketPrice(uint marketId) public view returns (uint) {
```
Comment (yaronvel) [inline src/BCdpManager.t.sol:167]: better testing here would be to revert/return 0 if market id is not 3

File: src/deployment/deployment.t.sol [hunk 2]
```diff
@@ -96,7 +96,8 @@ contract VatDeployer {
 
         spotter.poke("ETH-A");
 
-        pool = new Pool(address(vat), address(0x12345678), address(spotter), address(new FakeJug()));
+        address fakeDaiToUsdPriceFeed = address(0);
```
Comment (yaronvel) [inline src/deployment/deployment.t.sol]: this should be fixed, but let's leave it for now, will open an issue

File: src/pool/Pool.sol [hunk 3]
```diff
@@ -316,8 +326,18 @@ contract Pool is Math, DSAuth, LibNote {
         // update user rad
         rad[msg.sender] = sub(rad[msg.sender], sub(radBefore, radAfter));
 
-        uint userInk = mul(dink, shrn) / shrd;
-        dMemberInk = sub(dink, userInk);
+        // DAI to USD rate, scale 1e18
+        uint d2uPrice = dai2usd.getMarketPrice(3); // marketId = 3
```
Comment (yaronvel) [inline src/pool/Pool.sol]: 3 should probably be a constant here

File: src/pool/Pool.sol [hunk 4]
```diff
@@ -316,8 +326,18 @@ contract Pool is Math, DSAuth, LibNote {
...
+
+        // dMemberInk = debt * 1.065 * d2uPrice
+        // dMemberInk = dink * (shrn/shrd) * (d2uPrice/1e18)
+        dMemberInk = mul(mul(dink, shrn), d2uPrice) / mul(shrd, uint(1e18));
```
Comment (yaronvel) [inline src/pool/Pool.sol]: do we have a WAD constant here for 1e18? maybe user (1 ether)?

File: src/pool/Pool.t.sol [hunk 5]
```diff
@@ -1221,6 +1231,169 @@ contract PoolTest is BCdpManagerTestBase {
         pool.emergencyExecute(address(d), data);
     }
 
+    function testSetDaiToUsdPriceFeed() public {
+        address oldPriceFeed = address(pool.dai2usd());
```
Comment (yaronvel) [inline src/pool/Pool.t.sol:1235]: addition check here should be about the actual value.
set different price and see that it return different price

File: src/pool/Pool.t.sol [hunk 6]
```diff
@@ -1221,6 +1231,169 @@ contract PoolTest is BCdpManagerTestBase {
...
+            cdpCushion; //shh
+            winners; //shh
+            assertEq(bite[i], 26 ether);
+            assertEq(pool.rad(address(members[i])), (1000 ether - 50 ether * i - 26 ether) * RAY - 1);
+        }
+
+        // jar should get 2% from 104 * 1.1 / 130
+        assertEq(vat.gem("ETH", address(jar)), expectedInJar);
+    }
+
```
Comment (yaronvel) [inline src/pool/Pool.t.sol:1419]: i think very low rate should be ok. but nevertheless could you please add a test for 1 DAI = 0.8 USD?

**Review by yaronvel — CHANGES_REQUESTED** [review]

File: src/deployment/deployment.t.sol (hunk 2 above)
Comment (jchittoda) [reply src/deployment/deployment.t.sol]: anw fixed it

**Review by yaronvel — APPROVED** [review]

**Pull request merged**